smarts/core/default_map_builder.py

```diff
@@ -58,23 +59,19 @@ def get_road_map(map_spec) -> Tuple[RoadMap, str]:
         _existing_map = None
         gc.collect()
 
-    map_path = map_spec.source
-    if not os.path.isfile(map_path):
+    if not os.path.isfile(map_spec.source):
         map_path = os.path.join(map_spec.source, "map.net.xml")
         if not os.path.exists(map_path):
             raise FileNotFoundError(
                 f"Unable to find map in map_source={map_spec.source}."
             )
+        map_spec = replace(map_spec, source=map_path)
```

Comment:
Why is there a need to use `dataclasses.replace` here? Why can't you use `map_spec.source = map_path`?

Reply:
We could probably allow this given our current implementations, but it's usually considered "bad practice" to change a data object being passed to you b/c it changes it in the caller too, which the caller might not expect (unless it's documented that way).  So that's why I'm avoiding it here.